src/Adapter/Preferences/PreferencesConfiguration.php

```diff
@@ -70,24 +71,59 @@ public function getConfiguration()
      */
     public function updateConfiguration(array $configuration)
     {
-        if ($this->validateConfiguration($configuration)) {
-            $this->configuration->set('PS_SSL_ENABLED', $configuration['enable_ssl']);
-            $this->configuration->set('PS_SSL_ENABLED_EVERYWHERE', $configuration['enable_ssl_everywhere']);
-            $this->configuration->set('PS_TOKEN_ENABLE', $configuration['enable_token']);
-            $this->configuration->set('PS_ALLOW_HTML_IFRAME', $configuration['allow_html_iframes']);
-            $this->configuration->set('PS_USE_HTMLPURIFIER', $configuration['use_htmlpurifier']);
-            $this->configuration->set('PS_PRICE_ROUND_MODE', $configuration['price_round_mode']);
-            $this->configuration->set('PS_ROUND_TYPE', $configuration['price_round_type']);
-            $this->configuration->set('PS_DISPLAY_SUPPLIERS', $configuration['display_suppliers']);
-            $this->configuration->set('PS_DISPLAY_MANUFACTURERS', $configuration['display_manufacturers']);
-            $this->configuration->set('PS_DISPLAY_BEST_SELLERS', $configuration['display_best_sellers']);
-            $this->configuration->set('PS_MULTISHOP_FEATURE_ACTIVE', $configuration['multishop_feature_active']);
-            $this->configuration->set('PS_SHOP_ACTIVITY', $configuration['shop_activity']);
+        if (false === $this->validateConfiguration($configuration)) {
+            return [
+                [
+                    'key' => 'Invalid configuration',
+                    'domain' => 'Admin.Advparameters.Notification',
```

Comment:
Hi, this string already exists in **Admin.Notifications.Warning**
I think we can keep this one instead of creating a new one.